Doc updates - passthrough metrics, and chaining description (#7314)
solo-io/gloo: docs/content/guides/security/auth/extauth/passthrough_auth/grpc/_index.md

EOF
{{< /highlight >}}

{{% notice note %}}

> Passthrough services also allow for failing "open" through its [`failureModeAllow`]({{< versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto.sk/#settings" >}}) field. 
> By setting this field to `true`, the auth service responds with an `OK` if your server has a `5XX` response, or the request times out.

Reviewer: ```suggestion
Passthrough services also allow for failing "open" through the [`failureModeAllow`]({{< versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto.sk/#settings" >}}) field. 
By setting this field to `true`, the auth service responds with an `OK` if either your server returns a `5XX` response or the request times out.
```

> Passthrough services also allow for failing "open" through its [`failureModeAllow`]({{< versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto.sk/#settings" >}}) field. 
> By setting this field to `true`, the auth service responds with an `OK` if your server has a `5XX` response, or the request times out.

Reviewer: ```suggestion
Passthrough services also allow for failing "open" through the [`failureModeAllow`]({{< versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto.sk/#settings" >}}) field. 
By setting this field to `true`, the auth service responds with an `OK` if either your server returns a `5XX` response or the request times out.
```

> By setting this field to `true`, the auth service responds with an `OK` if either your server returns a `5XX` response or the request times out.

Reviewer 2: Since this is a gRPC service, should we clarify that it's a 5xx equivalent response? Is it worth it calling out which specific RPC codes result in an OK?

Author: i think there's multiple codes which could bloat this. `5XX-equivalent` works as well, imo!

"logger":"extauth","caller":"runner/run.go:179","msg":"got new config"
```

> ### Metrics

Reviewer 2: I like this addition about metrics. 2 thoughts/ideas on this:
1. If we want to explicitly call out individual metrics, can you also include the other passthrough metrics that are available?
2. Perhaps an alternative to defining them explicitly in the docs, we could instead expose a way for users to inspect the metrics, and then call out very specific / important metrics that we recommend alerting on for production installations. Are you familiar with how a user could inspect the extauth service and identify the full set of metrics? If not, lets find some people on the team tomorrow to chat about this as its a valuable debugging tool

Author: I'm not 100% familiar to how we can expose them (assuming expose means generating docs from the ext-auth code); also not familiar where/how users can inspect metrics

* Failure Mode Allow

>   * Metric Name: extauth.solo.io/passthrough_failure_bypass
>   * Description: The number of times a server error or timeout occurred and was bypassed through failure_mode_allow=true

Reviewer: ```suggestion
  * Metric Name: `extauth.solo.io/passthrough_failure_bypass`
  * Description: The number of times a server error or timeout occurred and was bypassed through the `failure_mode_allow=true` setting
```
